nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1059,11 +1057,6 @@ def build_train_valid_test_datasets(self):
             test_iters * global_batch_size,
         ]
 
-        if self.trainer.limit_val_batches <= 1.0 and isinstance(self.trainer.limit_val_batches, float):
-            train_valid_test_num_samples[
-                1
-            ] = 1  # This is to make sure we only have one epoch on every validation iteration
```

Comment:
Because of this line, validation never got done when a float value was passed since the number of validation samples was forced to 1, leading to `len(val_dataloader)` being 0. Hence removing it.